services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/ProjectSettingsManagerImpl.java

```diff
@@ -194,11 +193,8 @@ public void deleteProjectSetting(UserInfo userInfo, String id) throws DatastoreE
 		if (storageLocationSetting instanceof ExternalS3StorageLocationSetting) {
 			UserProfile userProfile = userProfileManager.getUserProfile(userInfo.getId().toString());
 			ExternalS3StorageLocationSetting externalS3StorageLocationSetting = (ExternalS3StorageLocationSetting) storageLocationSetting;
-			if (!StringUtils.isEmpty(externalS3StorageLocationSetting.getEndpointUrl())
-					&& !externalS3StorageLocationSetting.getEndpointUrl().equals("https://" + Constants.S3_HOSTNAME)) {
-				throw new NotImplementedException("Synapse does not yet support external S3 buckets in non-us-east-1 locations");
-			}
 			validateOwnership(externalS3StorageLocationSetting, userProfile);
+			validateBucketAccess(externalS3StorageLocationSetting);
```

Comment:
this should occur before `validateOwnership()` otherwise, `validateOwnership()` will call `s3client.getObject()`, which assumes US_EAST_1 when region not present, and get one of the errors at line 325 instead of a error telling users to check their bucket's permission.

Reply:
That's a logical assumption, @zimingd but it's actually not true.  There is a bucket, "synodos.eu.frankfurt", which is in a European region but from which Synapse can list and read objects (files) using a us-east-1 S3 client.  However, Synapse cannot detect the region and will generate broken presigned URLs when using the us-east-1 S3 client.  So the checks about downloading file and about detecting the region are independent/orthogonal.